pkg/networkservice/chains/nsmgr/suite_test.go

```diff
@@ -458,15 +486,37 @@ func (s *nsmgrSuite) Test_PassThroughLocalUsecase() {
 
 	var nsesReg [nsesCount]*registry.NetworkServiceEndpoint
 	for i := 0; i < nsesCount; i++ {
+		requestCount := 0
+
 		var additionalFunctionality []networkservice.NetworkServiceServer
 		if i != 0 {
+			connCtx := &networkservice.ConnectionContext{
+				ExtraContext: map[string]string{
+					fmt.Sprintf("testkey-%v", i): fmt.Sprintf("testvalue-%v", i),
+				},
+				IpContext: &networkservice.IPContext{},
+			}
+
 			additionalFunctionality = []networkservice.NetworkServiceServer{
 				chain.NewNetworkServiceServer(
 					clienturl.NewServer(s.domain.Nodes[0].NSMgr.URL),
+					checkrequest.NewServer(t, func(t *testing.T, request *networkservice.NetworkServiceRequest) {
+						if requestCount > 0 {
+							assert.Equal(
+								t,
+								connCtx.String(),
+								request.GetConnection().Context.String(),
+							)
+						}
+						requestCount++
+					}),
```

Comment:
It looks like it checks context on server forward reqest. We need to verify the request on the server backward.
Should we also check that context is expected on the client backward?
For example, we can add this check at line 516.